theseus/optimizer/manifold_gaussian.py

```diff
@@ -98,5 +98,42 @@ def update(
                 f"Tried to update using tensor on device: {precision.dtype} but precision "
                 f"is on device: {self.device}."
             )
+        if not torch.allclose(precision, precision.transpose(1, 2)):
+            raise ValueError("Tried to update precision with non-symmetric matrix.")
 
         self.precision = precision
+
+
+def local_gaussian(
```

Comment:
Should these be class methods as well? I am wondering if calling this just `local` is okay then?

Reply:
The should really be class methods of the `LieGroup` / `Manifold` class as then they would follow the same format as the existing `local` and `retract` functions. I think it wouldn't make sense to make this a class method of `ManifoldGaussian`. 

I initially added `local_gaussian` and `retract_gaussian` as class methods of `Manifold` and `LieGroup` but removed them as it caused a circular import. Maybe it is fine to just leave them here for now?

Reply:
Makes sense. 
How about we move them to be class functions with names local and retract. If you then continue using the class functions it will be easier to figure out later if there is a way to combine with the Manifold local/retract without needing changes everywhere.

Reply:
Oh, sorry, in my previous comment/question I thought that you had added `local_gaussian` as a class method of `ManifoldGaussian` (instead of `Manifold`).

Reply:
I don't think it makes sense to add these as class functions to the `ManifoldGaussian` class. It would be confusing as they would not be consistent with the existing local and retract functions in the `Manifold` class. 

e.g. `var1.local(var2)` gives the value of `var2` in the tangent plane at `var1` where both are `Manifold` instances.
`local_gaussian(var1, gauss1)` gives the mean and precision of `gauss1` in the tangent plane at `var1`, so it wouldn't make sense to make it a class function of `ManifoldGaussian`.